File: 2. Linear Data Structures/STUPMACH.py
def maximum_tokens(n, capacities):
    capacities.sort(reverse=True) 
    total_tokens = 0
    
    for i in range(n):
        total_tokens += min(capacities[i], i+1) 
    
    return total_tokens

t = int(input())


for _ in range(t):
    n = int(input()) 
    capacities = list(map(int, input().split())) 
    result = maximum_tokens(n, capacities)
    print(result)

File: 2. Linear Data Structures/STUPMACH.cpp
#include <iostream>
#include <vector>
#include <algorithm>
using namespace std;

int maximumTokens(int n, vector<int>& capacities) {
    sort(capacities.rbegin(), capacities.rend());
    int totalTokens = 0;
    
    for (int i = 0; i < n; ++i) {
        totalTokens += min(capacities[i], i+1);
    }
    
    return totalTokens;
}

int main() {
    int t;
    cin >> t;
    
    for (int i = 0; i < t; ++i) {
        int n;
        cin >> n;
        
        vector<int> capacities(n);
        for (int j = 0; j < n; ++j) {
            cin >> capacities[j];
        }
        
        int result = maximumTokens(n, capacities);
        cout << result << endl;
    }
    
    return 0;
}
